### ISSUE TITLE
too many @ in the result

### ISSUE THREAD

**Issue by kFoodie** [rsennrich/subword-nmt#71]
I trained two vocabularies with about 900M Chinese-English materials, and then coded two data sets (900M training set and 500K test set) with these two Chinese-English vocabularies.

The training set can get normal results, but there are many @ in the test set.

Before using subword-nmt for bpe, I had participled Chinese.

The corresponding instructions are as follows:

> python learn_joint_bpe_and_vocab.py --input data/train.en data/train.zh -s 32000 -o data/bpe32k --write-vocabulary data/vocab.en data/vocab.zh
python apply_bpe.py --vocabulary data/vocab.en --vocabulary-threshold 50 -c data/bpe32k < data/train.en > data/corpus.32k.en
python apply_bpe.py --vocabulary data/vocab.zh --vocabulary-threshold 50 -c data/bpe32k < data/train.zh > data/corpus.32k.zh
python apply_bpe.py --vocabulary data/vocab.zh --vocabulary-threshold 50 -c data/bpe32k < data/valid.zh > data/aval_bpe_enzh.zh
python apply_bpe.py --vocabulary data/vocab.en --vocabulary-threshold 50 -c data/bpe32k < data/valid.en > data/aval_bpe_enzh.en

The result is shown in the figure.

![image](https://user-images.githubusercontent.com/16017418/51721939-b1168a00-208e-11e9-8634-046edb963874.png)
![image](https://user-images.githubusercontent.com/16017418/51721981-becc0f80-208e-11e9-8166-50dec7565d51.png)

I don't know where the problem is. Please help me to answer it. Thank you very much.

**Comment by rsennrich** [comment]
Hey kFoodie,

it looks like you didn't tokenize or word-segment the text before applying subword segmentation. We usually tokenize English text with the Moses tokenizer first, and segment Chinese text with Jieba.

**Comment by kFoodie** [comment]
emmm……
I have no problem...
Thank you!!

**Comment by rsennrich** [comment]
glad to hear it's working fine.

**Comment by BattsetsegB** [comment]
hey, i have same problem,but i used to Mongolian in about 25k Chinese-Mongolian parallel sentence . i tokenized Mongolian and Chinese both text with the Moses tokenizer first, and segment  Chinese text with Jieba before applying subword segmentation and i used to preprocessing scripts in http://data.statmt.org/wmt17_systems/training/. When tokenizing it's warning: 
Tokenizer Version 1.1
Language: zh
Number of threads: 1
Tokenizer Version 1.1
Language: mn
Number of threads: 1
WARNING: No known abbreviations for language 'mn', attempting fall-back to English version...
Tokenizer Version 1.1
Language: zh
Number of threads: 1
Tokenizer Version 1.1
Language: mn
Number of threads: 1
WARNING: No known abbreviations for language 'mn', attempting fall-back to English version...
clean-corpus.perl: processing ./../data/corpus.tok.zh & .mn to ./../data/corpus.tok.clean, cutoff 1-80, ratio 9
..
Input sentences: 22319  Output sentences:  22186
Building prefix dict from the default dictionary ...
Dumping model to file cache /tmp/jieba.cache
Loading model cost 1.535 seconds.
Prefix dict has been built succesfully.
Building prefix dict from the default dictionary ...
Loading model from cache /tmp/jieba.cache
Loading model cost 0.190 seconds.
Prefix dict has been built succesfully.
no pair has frequency >= 2. Stopping
Processing ./../data/corpus.bpe.zh
Done
Processing ./../data/corpus.bpe.mn
Done

Mongolian text has too many @@ like this.left is orginal text, middle is tokenized and left
![orig_tok_out](https://user-images.githubusercontent.com/44496085/53737206-d638bc80-3ec6-11e9-866b-a07370fac20f.png)
 is result.
You think its look like didn't tokenized?Please help me to answer it. Thank you very much.

**Comment by rsennrich** [comment]
The preprocessing scripts in http://data.statmt.org/wmt17_systems/training/ define a minimum frequency threshold for subword units; only units with a frequency > 50 are allowed, and the rest are segmented into smaller units. With a training corpus of 25k sentences, it is possible that only few units meet this criterion.

There might also be other problems though, like an inconsistent use of Cyrillic and Latin characters.

**Comment by kexinhuang12345** [comment]
Hey, I am working on applying BPE and finding subword units for chemical string format. So there is no tokenization method for it, and each input word is a long string and I want to decompose it into many subword units. The result using your script has many @@. So correct me if I am wrong here, the @@ tells us where to segment the entire string based on the learned vocabulary? So it is not a problem? Thank you so much!!!
